### PR TITLE
Support displaying the popup when there is only one exact match

### PR DESCRIPTION
**Pull request by galeo**
As discussed in #396, a `show` value is added to `corfu-on-exact-match`, and it handles both auto and manual completion.

Please have a look. Thanks.

### CODE REVIEW

File: corfu.el [hunk 1]
```diff
@@ -907,10 +908,12 @@ Lookup STR in CANDS to restore text properties."
...
              ;; If completion is finished and cannot be further completed,
+             ;; and the value of corfu-on-exact-match is not 'show,
              ;; return 'finished. Otherwise setup the Corfu popup.
-             (if (consp (completion-try-completion
-                         newstr table pred newpt
-                         (completion-metadata newstr table pred)))
+             (if (or (consp (completion-try-completion
+                             newstr table pred newpt
+                             (completion-metadata newstr table pred)))
+                     (eq corfu-on-exact-match 'show))
```
Comment (minad) [inline corfu.el:916]: Check corfu-on-exact-match before completion-try-completion for efficiency.

**Comment by minad** [comment]
Thanks, this looks simple enough! I will give it a try.

File: corfu.el [hunk 2]
```diff
@@ -889,12 +889,13 @@ Lookup STR in CANDS to restore text properties."
...
-           str 'finished
-           (alist-get 'corfu--candidates (corfu--recompute str pt table pred)))
-          t)
+      ((and (guard (not (eq corfu-on-exact-match 'show))) 't)
+       (goto-char end)
+       (corfu--message "Sole match")
+       (corfu--exit-function
+        str 'finished
+        (alist-get 'corfu--candidates (corfu--recompute str pt table pred)))
+       t)
```
Comment (minad) [inline corfu.el:898]: The behavior here does not seem to be right. When I complete manually and get an exact match (with corfu-on-exact-match=show), this branch is not taken, but also not the next one. The popup is not shown as it should.

Comment (galeo) [reply corfu.el:898]: Do you set `completion-cycle-threshold` to `nil`?

Comment (minad) [reply corfu.el:898]: Shouldn't matter?

Comment (galeo) [reply corfu.el:898]: Sorry. I misunderstood what you meant and the code has been updated to fix this issue.

Comment (minad) [reply corfu.el]: Thanks, looks all good now.

**Comment by minad** [comment]
Merged. I simplified the setup slightly in https://github.com/minad/corfu/commit/9cebcfedca0ab165fcfc8a56acc2d2513a02b9e3. Please give it a check if everything works as expected for you.

**Comment by galeo** [comment]
Did some quick testing and it works well, thanks!

**Pull request merged**